geode-core/src/main/java/org/apache/geode/internal/tcp/MsgStreamer.java

```diff
@@ -378,28 +366,25 @@ public void close() throws IOException {
 
   /** override OutputStream's write() */
   @Override
-  public void write(byte[] source, int offset, int len) {
-    // if (logger.isTraceEnabled()) {
-    // logger.trace(" bytes={} offset={} len={}", source, offset, len);
-    // }
-    if (this.overflowBuf != null) {
-      this.overflowBuf.write(source, offset, len);
+  public void write(byte @NotNull [] source, int offset, int len) {
```

Comment:
The placement of this `@NotNull` annotation seems a little odd. Should this instead be 
```
write(@NotNull byte[] source, int offset, int len)
```
Also, if that reordering doesn't fix the false positive LGTM warning, it can be suppressed using 
```
@SuppressWarnings("lgtm[java/inefficient-output-stream]")
```

Reply:
It is a bug in LGTM. An issue was opened a year ago that they haven't resolved yet. I like the annotation though.